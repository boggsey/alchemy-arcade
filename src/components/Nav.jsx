import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Nav = ({ isAuthenticated, receiveLogout }) => (
  <nav>
    <Link to="/">Home</Link>
    {
      isAuthenticated && (
        <React.Fragment>
          <Link to="/roster">Roster</Link>
          <Link to="/roster" onClick={receiveLogout}>Logout</Link>
        </React.Fragment>
      )
    }
  </nav>
);

Nav.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  receiveLogout: PropTypes.func.isRequired,
};

export default Nav;
