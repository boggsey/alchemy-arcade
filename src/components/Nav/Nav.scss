@import "../../styles/variables.scss";

nav {
  position: absolute;
  z-index: 9994;
  top: 20px;
  left: 10px;

  a {
    color: white;
    display: inline-block;
    padding-right: 5px;
    color: $alchemy-green;
    font: 600 14px/18px 'Arial';
    text-transform: uppercase;
    text-decoration: none;
  }
}
