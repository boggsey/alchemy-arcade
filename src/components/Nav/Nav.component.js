import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './Nav.scss';

const Nav = ({ isAuthenticated, receiveLogout }) => (
  <nav>
    {
      isAuthenticated && (
        <React.Fragment>
          <Link to="/roster">Roster</Link>
          <Link to="/player/new">Add New</Link>
          <Link to="/" onClick={receiveLogout}>Logout</Link>
        </React.Fragment>
      )
    }
  </nav>
);

Nav.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  receiveLogout: PropTypes.func.isRequired,
};

export default Nav;
