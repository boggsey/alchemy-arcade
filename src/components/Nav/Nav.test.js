import React from 'react';
import { shallow } from 'enzyme';
import Nav from './Nav.component';

describe('Nav', () => {
  it('Hasn\'t changed since the last snapshot', () => {
    const wrapper = shallow(<Nav isAuthenticated receiveLogout={() => {}} />);

    expect(wrapper).toMatchSnapshot();
  });
});
