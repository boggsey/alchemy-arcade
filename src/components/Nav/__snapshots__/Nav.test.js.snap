// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Hasn't changed since the last snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    isAuthenticated={true}
    receiveLogout={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <React.Fragment>
        <Link
          replace={false}
          to="/roster"
        >
          Roster
        </Link>
        <Link
          replace={false}
          to="/player/new"
        >
          Add New
        </Link>
        <Link
          onClick={[Function]}
          replace={false}
          to="/"
        >
          Logout
        </Link>
      </React.Fragment>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/roster"
          >
            Roster
          </Link>,
          <Link
            replace={false}
            to="/player/new"
          >
            Add New
          </Link>,
          <Link
            onClick={[Function]}
            replace={false}
            to="/"
          >
            Logout
          </Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Roster",
            "replace": false,
            "to": "/roster",
          },
          "ref": null,
          "rendered": "Roster",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Add New",
            "replace": false,
            "to": "/player/new",
          },
          "ref": null,
          "rendered": "Add New",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Logout",
            "onClick": [Function],
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Logout",
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <React.Fragment>
          <Link
            replace={false}
            to="/roster"
          >
            Roster
          </Link>
          <Link
            replace={false}
            to="/player/new"
          >
            Add New
          </Link>
          <Link
            onClick={[Function]}
            replace={false}
            to="/"
          >
            Logout
          </Link>
        </React.Fragment>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/roster"
            >
              Roster
            </Link>,
            <Link
              replace={false}
              to="/player/new"
            >
              Add New
            </Link>,
            <Link
              onClick={[Function]}
              replace={false}
              to="/"
            >
              Logout
            </Link>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Roster",
              "replace": false,
              "to": "/roster",
            },
            "ref": null,
            "rendered": "Roster",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Add New",
              "replace": false,
              "to": "/player/new",
            },
            "ref": null,
            "rendered": "Add New",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Logout",
              "onClick": [Function],
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Logout",
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
